### PR TITLE
Improve PySAML2 documentation - Add undocumented configuration options

### PR DESCRIPTION
**Pull request by sebulibah**
### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Have you added an explanation of what problem you are trying to solve with this PR?
* [ ] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?
* [ ] Does your submission pass tests?
* [ ] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

File: docs/howto/config.rst [hunk 1]
```diff
@@ -508,6 +659,12 @@ sign_response
 Specifies if the IdP should sign the authentication response or not. Can be
 True or False. Default is False.
 
+want_authn_requests_signed
+""""""""""""""""""""""""""
+
+Indicates that the AuthnRequest received by this IdP should be signed. Can be True or False.
```
Comment (spaceone) [inline docs/howto/config.rst:665]:
```suggestion
Indicates that the AuthnRequest received by this IdP should be signed. Can be ``True`` or ``False``.
```

File: docs/howto/config.rst [hunk 2]
```diff
@@ -508,6 +659,12 @@ sign_response
...
+The default value is False.
```
Comment (spaceone) [inline docs/howto/config.rst:666]:
```suggestion
The default value is ``False``.
```

File: docs/howto/config.rst [hunk 3]
```diff
@@ -607,6 +764,79 @@ regular expressions.::
 
 Here only mail addresses that end with ".umu.se" will be returned.
 
+scope
+"""""
+
+A list of string values that will be used to set the ``<Scope>`` element
+The default value of regexp is ``false``.
```
Comment (spaceone) [inline docs/howto/config.rst:771]:
```suggestion
The default value of regexp is ``False``.
```

**Comment by c00kiemon5ter** [comment]
@sebulibah thanks! this is really nice. I think the work here will be _very_ useful for the users of the library.

The PR is marked a draft. I will be keeping an eye for when it's ready and go over it. Thanks again!

File: docs/howto/config.rst [hunk 4]
```diff
@@ -247,6 +247,32 @@ The globally unique identifier of the entity.
...
+description
+^^^^^^^^^^^
+
+A string value that sets the description of the PySAML2 entity.
+
+Example::
+
+    "description": "My IdP",
+
+verify_ssl_cert
```
[the remark this answers is not in the document]
Comment (sebulibah) [reply docs/howto/config.rst:268]: I have made some changes to this.

File: docs/howto/config.rst [hunk 5]
```diff
@@ -283,6 +324,26 @@ Indicates which certificates will be used for encryption capabilities::
         },
     ],
 
+generate_cert_info
```
[the remark this answers is not in the document]
Comment (sebulibah) [reply docs/howto/config.rst:328]: and this.

**Pull request merged**